src/items/item.js

```diff
@@ -140,36 +141,40 @@ export class Item extends ClickableObject {
     }
 
     setNumItems(num) {
-        this.numItems = num;
-
-        // Update text
-        if (this.numItemsText != undefined) {
-            let visualNum = "0";
-            let fillColor = "";
-
-            // Set format/color based on the amount
-            switch (true) {
-                case num < 1000:
-                    visualNum = num;
-                    fillColor = "orange";
-                    break;
-                case num < 10000:
-                    visualNum = (num / 1000).toFixed(1) + " k";
-                    fillColor = "white";
-                    break;
-                default:
-                    visualNum = (num / 1000000).toFixed(1) + " m";
-                    fillColor = "green";
-                    break;
-            }
-            this.numItemsText.text = visualNum;
-            this.numItemsText.setFill(fillColor);
-
-            // Make visible if item is also visible
-            if (num <= 1) {
-                this.numItemsText.visible = false;
-            } else if (this.sprite.visible) {
-                this.numItemsText.visible = true;
+        if (num <= 0) {
+            this.destroy();
+        } else {
+            this.numItems = num;
+
+            // Update text
+            if (this.numItemsText != undefined) {
+                let visualNum = "0";
+                let fillColor = "";
+
+                // Set format/color based on the amount
+                switch (true) {
+                    case num < 1000:
+                        visualNum = num;
+                        fillColor = "orange";
+                        break;
+                    case num < 10000:
```

Comment:
White should be less than 10m and only have the 'k' if > 100k and the 'm' if > 10m. Also I don't think there is a space between the number and k/m

Reply:
Looks like this also shows the decimal part (1.1k instead of 1100). If it has the k/m cutoff it should not show the decimals. I think the total amount of the stack should be visible via examine

Reply:
Should we add this to the fonts constant?

Reply:
Adding it to the fonts constant would be good.

Reply:
Lemme double check on this:

- 0-999: orange, show full # 
- 1000 - 99,999: white, show full # 
- 100k - 9999k: white, truncate and show k (this seems weird)
- 10m and above: green

Reply:
0-9999: orange, show full # 
100k-9999k: white, truncate and show k
10m- 2147483647 (2147m, max signed 32bit): green, truncate and show m